### PR TITLE
Batcher

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

File: src/main/java/com/nesscomputing/callback/BatchingCallback.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    private final Callback<? super List<T>> out;
+    private final int size;
+
+    private BatchingCallback(int size, Callback<? super List<T>> out)
+    {
+        if (size <= 0) {
+            throw new IllegalArgumentException("Size must be positive, was " + size);
+        }
+        if (out == null) {
+            throw new IllegalArgumentException("Null callback");
```
Comment (leejaywu) [inline src/main/java/com/nesscomputing/callback/BatchingCallback.java:48]: ...ah, 'k, ArrayBlockingQueue should itself throw an IllegalArgumentException if somebody is being silly and specifies a size of 0.

File: src/main/java/com/nesscomputing/callback/BatchingCallback.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) 2012 Ness Computing, Inc.
+ *
```
Comment (leejaywu) [inline src/main/java/com/nesscomputing/callback/BatchingCallback.java:3]: 2013.

File: src/main/java/com/nesscomputing/callback/BatchingCallback.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    void commitInternal() throws CallbackRefusedException
+    {
+        final List<T> outList = new ArrayList<T>(size);
+        list.drainTo(outList);
+        if (!outList.isEmpty()) {
+            try {
+                out.call(outList);
+            } catch (final CallbackRefusedException | RuntimeException e) {
```
Comment (leejaywu) [inline src/main/java/com/nesscomputing/callback/BatchingCallback.java:106]: Internal, but not private?

Comment (stevenschlansker) [inline src/main/java/com/nesscomputing/callback/BatchingCallback.java:106]: fixed

File: src/test/java/com/nesscomputing/callback/TestBatchingCallback.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) 2013 Ness Computing, Inc.
+ *
```
Comment (leejaywu) [inline src/test/java/com/nesscomputing/callback/TestBatchingCallback.java:3]: 2013 again.

**Comment by leejaywu** [comment]
Code looks sane.  Only minor nits -- the 2012 copyrights instead of 2013, and an internal method that's package-visible instead of either private or protected and AFAICT is not VisibleForTesting.

**Comment by stevenschlansker** [comment]
Fixed.

**Pull request merged**